lib/alm-cdk-challenge-stack.ts

```diff
@@ -1,16 +1,41 @@
-import { Stack, StackProps } from 'aws-cdk-lib';
+import { RemovalPolicy, Stack, StackProps, aws_s3 as S3 } from 'aws-cdk-lib';
+import { LambdaIntegration, RestApi } from 'aws-cdk-lib/aws-apigateway';
+import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
 import { Construct } from 'constructs';
-// import * as sqs from 'aws-cdk-lib/aws-sqs';
 
 export class AlmCdkChallengeStack extends Stack {
   constructor(scope: Construct, id: string, props?: StackProps) {
     super(scope, id, props);
 
-    // The code that defines your stack goes here
+    const bucket = new S3.Bucket(this, 'alm-cdk-challenge-bucket', {
+      bucketName: 'alm-cdk-challenge-bucket',
```

Comment:
Because you pass the bucket name to the function as an environment variable you can remove the bucketName parameter.
S3 buckets are a global resource in AWS so it is better to be careful and not create a bucket with a specific name (unfortunately in ops we are using an "hardcoded" name with a short hash to it )